File: iosApp/iosApp/Race/CircuitImage.swift
import SwiftUI

func circuitImageName(circuit: String, grandPrix: String = "") -> String? {
    for name in [circuit, grandPrix] {
        if let match = circuitImageKey(name) { return match }
    }
    return nil
}

private func circuitImageKey(_ name: String) -> String? {
    let key = name.lowercased()
    if key.contains("bahrain")    || key.contains("sakhir")      { return "Circuits/circuit_bahrain" }
    if key.contains("jeddah")     || key.contains("corniche")     { return "Circuits/circuit_jeddah" }
    if key.contains("albert")     || key.contains("melbourne")    { return "Circuits/circuit_australia" }
    if key.contains("suzuka")     || key.contains("japan")        { return "Circuits/circuit_japan" }
    if key.contains("shanghai")   || key.contains("china")        { return "Circuits/circuit_china" }
    if key.contains("miami")                                       { return "Circuits/circuit_miami" }
    if key.contains("imola")      || key.contains("ferrari")      { return "Circuits/circuit_imola" }
    if key.contains("monaco")                                      { return "Circuits/circuit_monaco" }
    if key.contains("barcelona")  || key.contains("catalunya")    { return "Circuits/circuit_spain" }
    if key.contains("villeneuve") || key.contains("canada")       { return "Circuits/circuit_canada" }
    if key.contains("red bull")   || key.contains("austria") || key.contains("spielberg") { return "Circuits/circuit_austria" }
    if key.contains("silverstone") || key.contains("britain")     { return "Circuits/circuit_silverstone" }
    if key.contains("hungaroring") || key.contains("hungary")     { return "Circuits/circuit_hungary" }
    if key.contains("spa")        || key.contains("belgium")      { return "Circuits/circuit_belgium" }
    if key.contains("zandvoort")  || key.contains("netherlands")  { return "Circuits/circuit_netherlands" }
    if key.contains("monza")      || key.contains("italy")        { return "Circuits/circuit_italy" }
    if key.contains("baku")       || key.contains("azerbaijan")   { return "Circuits/circuit_azerbaijan" }
    if key.contains("marina bay") || key.contains("singapore")    { return "Circuits/circuit_singapore" }
    if key.contains("americas")   || key.contains("cota") || key.contains("austin") { return "Circuits/circuit_usa" }
    if key.contains("hermanos")   || key.contains("mexico")       { return "Circuits/circuit_mexico" }
    if key.contains("interlagos") || key.contains("brazil") || key.contains("pace") { return "Circuits/circuit_brazil" }
    if key.contains("las vegas")  || key.contains("strip")        { return "Circuits/circuit_las_vegas" }
    if key.contains("losail")     || key.contains("lusail") || key.contains("qatar") { return "Circuits/circuit_qatar" }
    if key.contains("yas")        || key.contains("abu dhabi")    { return "Circuits/circuit_abu_dhabi" }
    if key.contains("madrid")                                      { return "Circuits/circuit_madrid" }
    return nil
}

struct CircuitImageView: View {
    let circuitName: String
    let grandPrixName: String
    let colors: AppColors
    var height: CGFloat = 100

    private var imageName: String? {
        circuitImageName(circuit: circuitName, grandPrix: grandPrixName)
    }

    var body: some View {
        ZStack {
            RoundedRectangle(cornerRadius: 8)
                .fill(colors.cardBorder.opacity(0.2))
            if let name = imageName {
                Image(name)
                    .resizable()
                    .scaledToFit()
                    .padding(4)
            } else {
                Text("🏁")
                    .font(.system(size: 24))
                    .opacity(0.3)
            }
        }
        .frame(maxWidth: .infinity)
        .frame(height: height)
    }
}

File: composeApp/src/androidMain/kotlin/org/gce/racehub/race/CircuitImage.kt
package org.gce.racehub.race

import androidx.annotation.DrawableRes
import org.gce.racehub.R

@DrawableRes
fun circuitDrawable(circuitName: String): Int? {
    val key = circuitName.lowercase()
    return when {
        "bahrain"    in key || "sakhir"    in key -> R.drawable.circuit_bahrain
        "jeddah"     in key || "corniche"  in key -> R.drawable.circuit_jeddah
        "albert"     in key || "melbourne" in key -> R.drawable.circuit_australia
        "suzuka"     in key || "japan"     in key -> R.drawable.circuit_japan
        "shanghai"   in key || "china"     in key -> R.drawable.circuit_china
        "miami"      in key                       -> R.drawable.circuit_miami
        "imola"      in key || "ferrari"   in key -> R.drawable.circuit_imola
        "monaco"     in key                       -> R.drawable.circuit_monaco
        "barcelona"  in key || "catalunya" in key -> R.drawable.circuit_spain
        "villeneuve" in key || "canada"    in key -> R.drawable.circuit_canada
        "red bull"   in key || "austria"   in key || "spielberg" in key -> R.drawable.circuit_austria
        "silverstone" in key || "britain"  in key -> R.drawable.circuit_silverstone
        "hungaroring" in key || "hungary"  in key -> R.drawable.circuit_hungary
        "spa"        in key || "belgium"   in key -> R.drawable.circuit_belgium
        "zandvoort"  in key || "netherlands" in key -> R.drawable.circuit_netherlands
        "monza"      in key || "italy"     in key -> R.drawable.circuit_italy
        "baku"       in key || "azerbaijan" in key -> R.drawable.circuit_azerbaijan
        "marina bay" in key || "singapore" in key -> R.drawable.circuit_singapore
        "americas"   in key || "cota"      in key || "austin" in key -> R.drawable.circuit_usa
        "hermanos"   in key || "mexico"    in key -> R.drawable.circuit_mexico
        "interlagos" in key || "brazil"    in key || "pace" in key -> R.drawable.circuit_brazil
        "las vegas"  in key || "strip"     in key -> R.drawable.circuit_las_vegas
        "losail"     in key || "lusail"    in key || "qatar"     in key -> R.drawable.circuit_qatar
        "yas"        in key || "abu dhabi" in key -> R.drawable.circuit_abu_dhabi
        "madrid"     in key                       -> R.drawable.circuit_madrid
        else -> null
    }
}
